pkg/util/deployment/deployment.go

```diff
@@ -236,6 +236,10 @@ func addHashKeyToRSAndPods(deployment *extensions.Deployment, c clientset.Interf
 		return updatedRS, nil
 	}
 
+	if err = waitForPodsHashPopulated(c, namespace, updatedRS.Name); err != nil {
+		return nil, fmt.Errorf("%s %s/%s: error waiting for controller to observe template hash label being added to pod label: %v", updatedRS.Kind, updatedRS.Namespace, updatedRS.Name, err)
```

Comment:
suggest `template hash label being added to pod label` -> `pods being labeled with template hash`